Tests/ConfidenceTests/ConfidenceValueTests.swift

```diff
@@ -0,0 +1,136 @@
+import Confidence
+import XCTest
+
+final class ConfidenceConfidenceValueTests: XCTestCase {
+    func testNull() {
+        let value = ConfidenceValue(null: ())
+        XCTAssertTrue(value.isNull())
+    }
+
+    func testIntShouldConvertToInt() {
+        let value = ConfidenceValue(integer: 3)
+        XCTAssertEqual(value.asInteger(), 3)
+    }
+
+    func testDoubleShouldConvertToDouble() {
+        let value = ConfidenceValue(double: 3.14)
+        XCTAssertEqual(value.asDouble(), 3.14)
+    }
+
+    func testBoolShouldConvertToBool() {
+        let value = ConfidenceValue(boolean: true)
+        XCTAssertEqual(value.asBoolean(), true)
+    }
+
+    func testStringShouldConvertToString() {
+        let value = ConfidenceValue(string: "test")
+        XCTAssertEqual(value.asString(), "test")
+    }
+
+    func testStringShouldConvertToDate() throws {
+        let formatter = DateFormatter()
+        formatter.dateFormat = "yyyy-MM-dd HH:mm:ss"
+        let date = try XCTUnwrap(formatter.date(from: "2022-01-01 12:00:00"))
+        let value = ConfidenceValue(timestamp: date)
+        XCTAssertEqual(value.asDate(), date)
+    }
+
+    func testStringShouldConvertToDateComponents() {
+        let dateComponents = DateComponents(year: 2024, month: 4, day: 3)
+        let value = ConfidenceValue(date: dateComponents)
+        XCTAssertEqual(value.asDateComponents(), dateComponents)
+    }
+
+    func testListShouldConvertToList() throws {
+        let formatter = DateFormatter()
+        formatter.dateFormat = "yyyy-MM-dd HH:mm:ss"
+        formatter.timeZone = TimeZone(abbreviation: "UTC")
+        let date1 = try XCTUnwrap(formatter.date(from: "2022-01-01 12:00:00"))
+        let dateComponents1 = DateComponents(year: 2024, month: 4, day: 3)
+        let date2 = try XCTUnwrap(formatter.date(from: "2022-01-02 00:00:00"))
+        let dateComponents2 = DateComponents(year: 2024, month: 4, day: 2)
+
+        let boolListValue = ConfidenceValue(boolList: [true, false])
+        let integerListValue = ConfidenceValue(integerList: [3, 4])
+        let doubleListValue = ConfidenceValue(doubleList: [3.14, 4.0])
+        let stringListValue = ConfidenceValue(stringList: ["val1", "val2"])
+        let timestampListValue = ConfidenceValue(timestampList: [date1, date2])
+        let dateListValue = ConfidenceValue(dateList: [dateComponents1, dateComponents2])
```

Comment:
Is it better to use arg labels `timestampList` and `dateList` (i.e. the Confidence-types), like shown here, or rather use respectively `dateList` and `dateComponentsList` (i.e. the accepted Swift types)?